hi-ml/testhiml/testhiml/test_run_ml.py

```diff
@@ -204,19 +252,78 @@ def _expected_files_exist() -> bool:
     assert _expected_files_exist()
 
 
-def test_run(ml_runner_with_container: MLRunner) -> None:
+@pytest.mark.parametrize("run_extra_val_epoch", [True, False])
+@pytest.mark.parametrize("run_inference_only", [True, False])
+def test_run(run_inference_only: bool, run_extra_val_epoch: bool, ml_runner_with_container: MLRunner) -> None:
     """Test that model runner gets called """
+    ml_runner_with_container.container.run_inference_only = run_inference_only
+    ml_runner_with_container.container.run_extra_val_epoch = run_extra_val_epoch
     ml_runner_with_container.setup()
     assert not ml_runner_with_container.checkpoint_handler.has_continued_training
     with patch.object(ml_runner_with_container, "checkpoint_handler"):
```

Comment:
You may find it clearer to use [patch.multiple](https://docs.python.org/3/library/unittest.mock.html#patch-multiple) here for mocking many different methods of the same object